### ISSUE TITLE
Any idea why is so slow?

### ISSUE THREAD

**Issue by bokicans** [10up/wp-local-docker-v2#25]
I test this with empty WP and getting time to first bite (TTFB) around 600ms which, I believe, is slow.
With docker-php-ext-xdebug.ini turned on is almost 2sec.
In some other environments I'm getting TTFB in ~80-200ms, talking about empty WP installation.

My computer is Ryzen7 with 8 cores, 16gb RAM and M.2 SSD (read 3.5gb/s) so that cannot be an issue.
I like this concept and would like to find out where is the bottle neck.

note: I posted same question to beta repo first by mistake. (https://github.com/10up/wp-local-docker/issues/136)

thanks
Bojan

**Comment by cmmarslender** [comment]
Hoping that some of the changes in https://github.com/10up/wp-local-docker-v2/commit/c8580fd60ee55a56fc4e0ff5abb9d3d8e78fe234 help with this issue.

Once 2.4 is available, can you test and see if the speed is any better?

**Comment by WesternWeb** [comment]
I'm having the same issue

**Comment by dinhtungdu** [comment]
I can get ~300ms with xdebug disabled on my 2015 15" Macbook Pro with 16GB SSD (2G/s) running macOS 10.14.6.

With xdebug enabled, the TTFB is ~600ms.

**Comment by jeffpaul** [comment]
@bokicans @WesternWeb note that there have been a couple releases since you called out concerns on speed, have you updated to v2.5.1 and still seeing serious issues?  One thing to also note is that Docker is slow on its own, so there will be some limits to how fast WP Local Docker will be.

**Comment by RiFi2k** [comment]
Yeah pretty much this is because of your filesystem mounts, nothing they can do for this past the cache on the mounts which is what [docker recommends](https://docs.docker.com/docker-for-mac/osxfs-caching/).

Go ahead and get a desktop and install Fedora linux or some easy to use distro on it, then install all these tools and you will die when you see how much faster it is. Couple people here use windows and we can start the same webpack script in a docker container, which has to do a lot of work and write a ton of files at the same time and they will still be going 10+ minutes after mine is done. Same with just developing locally, mine is blazing theirs feels like 3G network speeds.

**Comment by dustinrue** [comment]
Two things. The updated mount options in the more recently releases of wp-local-docker-v2 will help AND a lot of the slow down is due to xdebug.

Hope that helps

**Comment by allysonsouza** [comment]
I'm trying to use WP Local Docker in Windows 10 and it's really slow. I guess it's because of the filesystem (using VCCW/VirtualBox before, changing to NFS had a great performance impact). 

I'm not a Docker specialist, but it seems possible to change the mount file system to NFS too, but it looks complex to my current knowledge: https://github.com/docker/for-win/issues/507

**Comment by bitfactory-lasse-vandenboschchristensen** [comment]
Hi all,

Maybe disabling xdebug will help you out (as per @dustinrue suggestion): https://github.com/10up/wp-local-docker-v2/issues/47#issuecomment-557100434

**Comment by junaidbhura** [comment]
Enabling OP Cache seems to have an impact for me. I’ve created some images based on 10up’s:
https://hub.docker.com/r/junaidbhura/phpfpm/tags

Can someone try using `junaidbhura/phpfpm:7.3` or `junaidbhura/phpfpm:7.2` to confirm?
